File: Tests/DungeonSoundboardCoreTests/RepeatModeTests.swift
import XCTest
@testable import DungeonSoundboardCore

final class RepeatModeTests: XCTestCase {
    func testFromStoredValueSupportsCurrentValues() {
        XCTAssertEqual(RepeatMode.fromStoredValue("off"), .off)
        XCTAssertEqual(RepeatMode.fromStoredValue("one"), .one)
        XCTAssertEqual(RepeatMode.fromStoredValue("all"), .all)
    }

    func testFromStoredValueSupportsLegacyLocalizedValues() {
        XCTAssertEqual(RepeatMode.fromStoredValue("Без повтора"), .off)
        XCTAssertEqual(RepeatMode.fromStoredValue("Повтор трека"), .one)
        XCTAssertEqual(RepeatMode.fromStoredValue("Повтор плейлиста"), .all)
    }

    func testLocalizedKeysAreStable() {
        XCTAssertEqual(RepeatMode.off.localizedKey, "repeat.off")
        XCTAssertEqual(RepeatMode.one.localizedKey, "repeat.one")
        XCTAssertEqual(RepeatMode.all.localizedKey, "repeat.all")
    }
}

File: windows/tests/DungeonSoundboard.Core.Tests/RepeatModeTests.cs
using DungeonSoundboard.Core.Models;
using Xunit;

namespace DungeonSoundboard.Core.Tests;

public sealed class RepeatModeTests
{
    [Fact]
    public void FromStoredValueSupportsCurrentValues()
    {
        Assert.Equal(RepeatMode.Off, RepeatModeExtensions.FromStoredValue("off"));
        Assert.Equal(RepeatMode.One, RepeatModeExtensions.FromStoredValue("one"));
        Assert.Equal(RepeatMode.All, RepeatModeExtensions.FromStoredValue("all"));
    }

    [Fact]
    public void FromStoredValueSupportsLegacyLocalizedValues()
    {
        Assert.Equal(RepeatMode.Off, RepeatModeExtensions.FromStoredValue("Без повтора"));
        Assert.Equal(RepeatMode.One, RepeatModeExtensions.FromStoredValue("Повтор трека"));
        Assert.Equal(RepeatMode.All, RepeatModeExtensions.FromStoredValue("Повтор плейлиста"));
    }

    [Fact]
    public void LocalizedKeysAreStable()
    {
        Assert.Equal("repeat.off", RepeatMode.Off.LocalizedKey());
        Assert.Equal("repeat.one", RepeatMode.One.LocalizedKey());
        Assert.Equal("repeat.all", RepeatMode.All.LocalizedKey());
    }
}
